pkg/controller/job/job_controller_test.go

```diff
@@ -675,6 +676,25 @@ func TestControllerSyncJob(t *testing.T) {
 			expectedFailed:        0,
 			expectedPodPatches:    5,
 		},
+		"indexed job with pod indexes higher than parallelism has higher indexes removed first": {
```

Comment:
```suggestion
		"indexed job with pod indexes higher than completions has higher indexes removed first": {
```

Reply:
but this is already the test case above "indexed job with indexes outside of range"

Reply:
oh this unit test was a remnant of the original "allow spec.Completions to be nil" proposal which used parallelism to scale the job while completions is nil. I have removed this test now.